mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -199,12 +199,12 @@ object MatrixFactorizationModel extends Loader[MatrixFactorizationModel] {
       assert(formatVersion == thisFormatVersion)
       val rank = (metadata \ "rank").extract[Int]
       val userFeatures = sqlContext.parquetFile(userPath(path))
-        .map { case Row(id: Int, features: Seq[Double]) =>
-          (id, features.toArray)
+        .map { case Row(id: Int, features: Seq[_]) =>
+          (id, features.asInstanceOf[Seq[Double]].toArray)
```

Comment:
Hm, I tried putting `@unchecked` just about everywhere it seemed to allow (method, statement, etc.) and it either didn't compile or didn't eliminate the warning. It does seem like this should be possible and right-er but there may be a reason that it would require more elaborate rewriting in order to employ it.